src/test/shared/settingsConfiguration.test.ts

```diff
@@ -0,0 +1,78 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import * as vscode from 'vscode'
+import { extensionSettingsPrefix } from '../../shared/constants'
+import { DefaultSettingsConfiguration } from '../../shared/settingsConfiguration'
+
+describe('DefaultSettingsConfiguration', () => {
+    // These tests use an actual extension setting, because vscode.WorkspaceConfiguration fails when
+    // you attempt to update one that isn't defined in package.json. We will restore the setting value
+    // at the end of the tests.
```

Comment:
Is there any risk that the `after()` call isn't made?

Also, is it choking on the fact that it's not in package.json or that it's not in the settings.json file in the workspace? If it's the latter, can we preload in a settings.json file?

Reply:
There's no risk for CI systems. For repo developers, its a mild irritant.

We might be able to leverage a workspace settings in the tests instead of global workspace. This test would have to move to the integration tests then, to have access to a workspace. I'll put that into the backlog.

Reply:
Note: the `settings.json` file isn't tied to the workspace; it's global for VS Code (for Windows, it's in `C:\Users\<USERNAME>\AppData\Roaming\Code\User\settings.json`).

Obviously, we wouldn't want to overwrite it, but since it is a JSON file, we could potentially append a dummy field.